src/components/MyAccount/Settings/AccountSettingsTab.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  Box,
+  Form,
+  List,
+  Spacer,
+  useModal,
+} from "@nypl/design-system-react-components"
+import { useState } from "react"
+import type { Patron } from "../../../types/myAccountTypes"
+import styles from "../../../../styles/components/MyAccount.module.scss"
+import AccountSettingsButtons from "./AccountSettingsButtons"
+import {
+  AccountSettingsForm,
+  AccountSettingsDisplay,
+} from "./AccountSettingsDisplayOptions"
+import {
+  successModalProps,
+  failureModalProps,
+} from "./AccountSettingsFeedbackModalProps"
+import { parsePayload, updatePatronData } from "./AccountSettingsUtils"
+import { BASE_URL } from "../../../config/constants"
+
+const AccountSettingsTab = ({ settingsData }: { settingsData: Patron }) => {
+  const [currentlyEditing, setCurrentlyEditing] = useState(false)
+  const [mostRecentPatronData, setMostRecentPatronData] = useState(settingsData)
+  const [modalProps, setModalProps] = useState(null)
+  const listElements = currentlyEditing ? (
+    <AccountSettingsForm patron={mostRecentPatronData} />
+  ) : (
+    <AccountSettingsDisplay patron={mostRecentPatronData} />
+  )
```

Comment:
I think this is what you flagged in the PR description. I would try to have one component that returns either the form or the display, but doing it this way prevents having to pass `currentlyEditing` into the single component. And this is also done at the parent level so this seems fine to me.

You mentioned it's a little kooky but is this also what you were referring to?

Reply:
Yea I like that it's happening in the parent. The kookiness I'm referring to is more about the business of like bending the icon list element pattern into the form. I feel like I have all of these switch statements everywhere and I'm not sure if it's legible to anyone but me.